consul/resource_consul_license_test.go

```diff
@@ -0,0 +1,28 @@
+package consul
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccConsulLicense_basic(t *testing.T) {
```

Comment:
Is it possible to add some kind of conditional test that runs against the enterprise version?

Reply:
Do you think it's OK if the test look for a `test_license.hcli` in the working dir and run if it finds one?

Reply:
Seems reasonable with documentation and logging. Another option is an API query to /v1/agent/self that maybe interrogates the version string or similar?

Reply:
I did both, some resources like the `consul_network_segment` or the namespace support can be run without actually having a license when using the `hashicorp/consul-enterprise` Docker image.